library/x509.c

```diff
@@ -1437,7 +1437,18 @@ int mbedtls_x509_parse_subject_alt_name(const mbedtls_x509_buf *san_buf,
                    san_buf, sizeof(*san_buf));
         }
         break;
+        /*
+         * IP address
+         */
+        case (MBEDTLS_ASN1_CONTEXT_SPECIFIC | MBEDTLS_X509_SAN_IP_ADDRESS):
+        {
+            memset(san, 0, sizeof(mbedtls_x509_subject_alternative_name));
```

Comment:
Please note that `x509_crt_check_san` added in https://github.com/Mbed-TLS/mbedtls/pull/7436 will only check an IP san if there's no `MBEDTLS_X509_SAN_DNS_NAME` match, therefore, an IP that got to here can be invalid. 
You probably have to use `mbedtls_x509_crt_parse_cn_inet_pton` here.

Reply:
In the certificate we have IP represented as 4(Ipv4) or 16(Ipv6) bytes. So we are simply taking what we have. `mbedtls_x509_crt_parse_cn_inet_pton` tries to parse IP represented as string.

Reply:
Should we check here that `san_buf->len` is either `4` or `16`?

Reply:
I think we can add this check here.